dtfs-central-api/src/helpers/get-keying-sheet-calculation-tfm-facility-values.test.ts

```diff
@@ -0,0 +1,195 @@
+import { convertTimestampToDate } from './convert-timestamp-to-date';
+
+jest.mock('../repositories/tfm-facilities-repo');
+
+describe('getKeyingSheetCalculationTfmFacilityValues', () => {
+  const findOneByUkefFacilityIdSpy = jest.spyOn(TfmFacilitiesRepo, 'findOneByUkefFacilityId');
+  const facilityId = '123';
+  const today = new Date();
+
+  const mockFacility = {
+    ...aTfmFacility(),
+    facilitySnapshot: {
+      ...aFacility(),
+      ukefFacilityId: facilityId,
+      interestPercentage: 5,
+      dayCountBasis: 365,
+      coverEndDate: today,
+      coverStartDate: today,
+      coverPercentage: 5,
+    },
+  };
+
+  const reportPeriod = {
+    start: { month: today.getMonth() + 1, year: today.getFullYear() },
+    end: { month: today.getMonth() + 1, year: today.getFullYear() },
+  };
+
+  const dateBeforeReportPeriodEnd = new Date(subDays(today, 1));
+
+  const mockFacilityWithAmendment = {
+    ...mockFacility,
+    amendments: [
+      {
+        ...aCompletedTfmFacilityAmendment(),
+        value: 350000,
+        effectiveDate: dateBeforeReportPeriodEnd.getTime(),
+        // 365 days after report period end
+        coverEndDate: new Date(addDays(today, 365)).getTime(),
+      },
+    ],
+  };
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('when a tfm facility with the supplied facility id cannot be found', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue(null);
+    });
+
+    it('should throw a NotFoundError', async () => {
+      await expect(getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod)).rejects.toThrow(
+        new NotFoundError(`TFM facility with ukefFacilityId '${facilityId}' could not be found`),
+      );
+    });
+  });
+
+  describe('when the found tfm facility does not contain a cover start date', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue({
+        ...aTfmFacility(),
+        facilitySnapshot: {
+          ...aFacility(),
+          ukefFacilityId: facilityId,
+          interestPercentage: 5,
+          dayCountBasis: 365,
+          coverEndDate: today,
+          coverStartDate: null,
+        },
+      });
+    });
+
+    it('should throw a NotFoundError', async () => {
+      // Act / Assert
+      await expect(getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod)).rejects.toThrow(
+        new NotFoundError(`Failed to find a cover start date for the tfm facility with ukef facility id '${facilityId}`),
+      );
+    });
+  });
+
+  describe('when the found tfm facility does not contain a cover end date', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue({
+        ...aTfmFacility(),
+        facilitySnapshot: {
+          ...aFacility(),
+          ukefFacilityId: facilityId,
+          interestPercentage: 5,
+          dayCountBasis: 365,
+          coverEndDate: null,
+          coverStartDate: today,
```

Comment:
Do we need all of these values, the test only cares about `coverEndDate` being null

Reply:
changed